contracts/src/NativeTokenDestination.sol

```diff
@@ -296,29 +288,56 @@ contract NativeTokenDestination is
     }
 
     /**
-     * @dev See {TeleporterTokenDestination-_scaleTokens}
+     * @dev See {TeleporterTokenDestination-_handleSendAndCall}
+     *
+     * Mints the tokens to this contract, and send them to the recipient contract as a
+     * part of the call to {INativeSendAndCallReceiver-receiveTokens} on the recipient contract.
+     * If the call fails, the amount is sent to the fallback recipient.
+     *
+     * Note: If the recipient contract does not properly handle the full msg.value sent,
+     * the balance can be locked in the recipient contract. Receiving contracts must make
+     * sure to properly handle the balance to ensure it does not get locked improperly.
      */
-    function _scaleTokens(uint256 value, bool isReceive) internal view override returns (uint256) {
-        // Multiply when multiplyOnReceive and isReceive are both true or both false.
-        if (multiplyOnReceive == isReceive) {
-            return value * tokenMultiplier;
+    function _handleSendAndCall(
+        SingleHopCallMessage memory message,
+        uint256 amount
+    ) internal virtual override {
+        // If the contract is not yet fully collateralized, the use of send and call is not allowed
+        // because it could result in unexpected behavior given that the amount of tokens used to make the
+        // call to "receiveTokens" is less than expected. Instead, the amount is handled as a normal bridge
+        // event to fallback recipient.
+        if (currentReserveImbalance > 0) {
+            _withdraw(message.fallbackRecipient, amount);
```

Comment:
The call to `_withdraw` will emit a `CollateralAdded` event, which I think is sufficient for this case since it can only ever occur prior to the destination contract being properly initialized with collateral.